From: Alexander Aring <aar@pengutronix.de>
Date: Fri, 26 Aug 2016 18:11:44 +0200
Subject: [PATCH] libfakeroot: Hide error from dlsym()

grabbed from debian package:

dlsym(), starting in glibc 2.24 actually reports errors. In our case,
we try to get ACL functions which are not in the glibc. This causes
failures in test suites, so hide those messages for non-debugging
purposes for now. It also makes the build logs annoying to read.
Author: Julian Andres Klode <juliank@ubuntu.com>
Origin: vendor
Bug-Debian: https://bugs.debian.org/830912
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: no
Last-Update: 2016-08-12

Signed-off-by: Alexander Aring <aar@pengutronix.de>
---
 libfakeroot.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/libfakeroot.c b/libfakeroot.c
index 4fd620f8a5f0..ca7494a97263 100644
--- a/libfakeroot.c
+++ b/libfakeroot.c
@@ -256,10 +256,16 @@ void load_library_symbols(void){
  /* clear dlerror() just in case dlsym() legitimately returns NULL */
     msg = dlerror();
     *(next_wrap[i].doit)=dlsym(get_libc(), next_wrap[i].name);
+
     if ( (msg = dlerror()) != NULL){
-      fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
-/*    abort ();*/
+#ifdef LIBFAKEROOT_DEBUGGING
+      if (fakeroot_debug) {
+        fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
+/*      abort ();*/
+      }
+#endif
     }
+
   }
 }
 
